### PR TITLE
feat: interactive components, improve focus outline

### PR DESCRIPTION
**Pull request by seedy**
## Description

- closes https://github.com/traefik/faency/issues/362

### Card

Now behaves like a button when interactive.

Outline color is "stronger" when element is focused.

https://user-images.githubusercontent.com/3367393/164496842-48e51e83-378e-4f04-9c1c-984bb5249590.mp4

### Badge

Outline is 2px rather than 1px, I felt it was not visible enough, before.

Styles were not correctly applied, therefore the outline depending on the `variant` value was not applied.

### Switch

Added outline around the button when it is focused. I felt elevation on the thumb was not visible enough.

### Misc

Some cleanup while checking components and their stories.

## How to test?

Check stories for updated components:
- `Badge`
- `Card`
- `Switch`

Their `interactive` version should be tabbable and their focused state should be visible enough.

Please note I made decisions alone. Please, feel free to give your own opinion during review :+1: 

## Good PR checkboxes

- [x] Change has been tested
- [ ] Added/Updated tests
- [x] Added/Updated stories
- [x] PR follows [conventions](https://github.com/traefik/faency#how-to-contribute)
- [x] Labels are set
- [x] Project is linked

## Good Review checkboxes

<details>
<summary> ℹ️  Copy the snippet and paste in the review field to fill it</summary>

```markdown
- [ ] I've tested the changes
- [ ] I've agreed on the unit tests (soon to come)
- [ ] I've checked the stories
- [ ] I've read the code and understood it
- [ ] I don't have any more questions
- [ ] I've described any optional improvements
- [ ] I checked PR follows [conventions](https://github.com/traefik/faency#how-to-contribute)
```

</details>

### CODE REVIEW

File: components/Badge/Badge.tsx [hunk 1]
```diff
@@ -5,13 +5,26 @@ import { Slot } from '@radix-ui/react-slot';
...
+);
+
+const interactiveColorVariants: ColorVariants = COLORS.reduce(
   (variants, color) => ({
     ...variants,
     [color]: {
-      bc: `$${color}6`,
-      color: `$${color}10`,
+      ...getColorBadgeStyles(color),
+      '&:focus-visible': { outline: `2px solid $${color}9` },
```
Comment (gndz07) [inline components/Badge/Badge.tsx:27]: What if we make border color similar accross components? Card and switch has the same `primary` color as border, but badge is not, wdyt?

image

image

**Review by gndz07 — APPROVED** [review]
LGTM, small comment.

File: components/Badge/Badge.tsx (hunk 1 above)
Comment (seedy) [reply components/Badge/Badge.tsx:27]: @gndz07 I don't have a strong opinion there. I just checked that primary color was fine with all the color variants.

I agree applying similar outline effect across components makes sense. I would simply apply the same `2px` outline on all those.

WDYT?

**Review by paulocfjunior — APPROVED** [review]

**Pull request merged**